### ISSUE TITLE
Long lists are no longer wrapped when rendering formatted SQL

### ISSUE THREAD

**Issue by masc3d** [jOOQ/jOOQ#12253]
### Your question:

Dumping sql using 3.14 `withRenderFormatted(true)` yields sql output

```
REPLACE INTO `tad_address` (
  `id`,
  `uid`,
  `activated`,
  `line1`,
  `line2`,
  `line3`,
  `phone_number`,
  `fax_number`,
  `mail`,
  `country_code`,
  `zip_code`,
  `city`,
  `street`,
  `street_no`,
  `latitude`,
  `longitude`,
  `geocode_invalid`,
  `sync_id`
) VALUES (
  19892371,
  X'c290e7b3e2f111ebaf8c0050568b4236',
  {ts '2021-07-21 00:00:00.0'},
  'Labor 28',
  null,
  ...
```

Starting with jooq-3.15 this switch doesn't seem to have any effect, sql is always rendered as a single line.

```
REPLACE INTO `dekuclient`.`tad_address` (`id`, `uid`, `activated`, `line1`, `line2`, `line3`, `phone_number`, `fax_number`, `mail`, ....
```

I appreciated the previous format, as it's much easier to read parameter assignments and some editors (eg datagrip)  perform significantly better. 
Is there any way to configure 3.15 to render the previous format?

**Comment by lukaseder** [comment]
Thanks for your report. I've noticed this as well in integration test logs. The change was introduced with https://github.com/jOOQ/jOOQ/issues/11672

There is a new concept of "simple query parts" (e.g. table / column references, bind values, etc.) and lists of such simple query parts. A `QueryPartCollectionView` is currently formatted only if it contains more than 1 elements *and* at least one non-simple element. This works decently for a lot of cases (see rationale and examples in https://github.com/jOOQ/jOOQ/issues/11672), but not so well for the simple case of listing columns in `INSERT` and other statements.

I'm not sure how to fix this yet. In the long run, there will be a lot of individual formatting options, similar to those in an IDE, see https://github.com/jOOQ/jOOQ/issues/6526. But this particular case can probably be solved more easily.

> some editors (eg datagrip) perform significantly better.

That's quite the bug in the editor then, which is worth reporting!

**Comment by lukaseder** [comment]
I guess there's a sweet spot after which a list of elements should be wrapped irrespective of whether the contents are "simple".

IntelliJ defaults to "keep elements on one line if <= 7":

![image](https://user-images.githubusercontent.com/734593/127124057-98c245e1-4b06-4a88-bbda-237da9eeca18.png)

I think that in the case of jOOQ, 4 might be a better default, keeping in mind the main use-case of the previous change having been functions like `DECODE()`, which are typically used with 4 arguments. It will be great to make this configurable in the future, via https://github.com/jOOQ/jOOQ/issues/6526. For now, in jOOQ 3.16.0 and 3.15.2, in order to fix this issue here, I'll implement a hard-coded "wrap elements if list is longer than 4" behaviour.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.2 (https://github.com/jOOQ/jOOQ/issues/12258)
